.github/workflows/publish-to-npm.yml

```diff
@@ -0,0 +1,95 @@
+name: publish-to-npm
+on:
+  release:
+    types: [released]
+  # Support manual releases in case something goes wrong, or we need to do a test.
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Tag to be published
+        type: string
+        required: true
+
+jobs:
+  # Step 1: Verify that the tag we're trying to release is a valid candidate for publishing.
+  verify-candidate-tag:
```

Comment:
This is pretty closely copied from `validate-candidate-branch.sh`, just split into a job and converted to use tags.